exporters/otlp/internal/transform/metric.go

```diff
@@ -0,0 +1,143 @@
+// Copyright 2020, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package transform provides translations for opentelemetry-go concepts and
+// structures to otlp structures.
+package transform
+
+import (
+	"errors"
+
+	commonpb "github.com/open-telemetry/opentelemetry-proto/gen/go/common/v1"
+	metricpb "github.com/open-telemetry/opentelemetry-proto/gen/go/metrics/v1"
+
+	"go.opentelemetry.io/otel/api/core"
+	metricsdk "go.opentelemetry.io/otel/sdk/export/metric"
+	"go.opentelemetry.io/otel/sdk/export/metric/aggregator"
+)
+
+// ErrUnimplementedAgg is returned when a transformation of an unimplemented
+// aggregator is attempted.
+var ErrUnimplementedAgg = errors.New("unimplemented aggregator")
+
+// Record transforms a Record into an OTLP Metric. An ErrUnimplementedAgg
+// error is returned if the Record Aggregator is not supported.
+func Record(r metricsdk.Record) (*metricpb.Metric, error) {
+	d := r.Descriptor()
+	l := r.Labels()
+	switch a := r.Aggregator().(type) {
+	case aggregator.MinMaxSumCount:
+		return minMaxSumCount(d, l, a)
+	case aggregator.Sum:
+		return sum(d, l, a)
+	}
+	return nil, ErrUnimplementedAgg
+}
+
+// sum transforms a Sum Aggregator into an OTLP Metric.
+func sum(desc *metricsdk.Descriptor, labels metricsdk.Labels, a aggregator.Sum) (*metricpb.Metric, error) {
+	sum, err := a.Sum()
+	if err != nil {
+		return nil, err
+	}
+
+	m := &metricpb.Metric{
+		MetricDescriptor: &metricpb.MetricDescriptor{
+			Name:        desc.Name(),
+			Description: desc.Description(),
+			Unit:        string(desc.Unit()),
+			Labels:      stringKeyValues(labels.Ordered()),
+		},
+	}
+
+	switch n := desc.NumberKind(); n {
+	case core.Int64NumberKind, core.Uint64NumberKind:
+		m.MetricDescriptor.Type = metricpb.MetricDescriptor_COUNTER_INT64
+		m.Int64Datapoints = []*metricpb.Int64DataPoint{
+			{Value: sum.CoerceToInt64(n)},
+		}
+	case core.Float64NumberKind:
+		m.MetricDescriptor.Type = metricpb.MetricDescriptor_COUNTER_DOUBLE
+		m.DoubleDatapoints = []*metricpb.DoubleDataPoint{
+			{Value: sum.CoerceToFloat64(n)},
+		}
+	}
+
+	return m, nil
+}
```

Comment:
Record with same metric descriptor but different label values could be grouped.
```go
m1 {
   descriptor: { ..}
   datapoints: []{
      {
        Labels: { {"k1", "v11"}. {"k2", "v21"}},
        value: 50
       },
      {
        Labels: { {"k1", "v12"}. {"k2", "v22"}},
        value: 500
       },
     }
}
```
It is only beneficial if there are large number of time-series for the same metrics.

Anyway, It doesn't need to be part of this PR.

Reply:
Agreed. I was planning to address batching in the "Add support for Resources" work on this. I'll be sure to link this comment with that Issue to track it.

Reply:
I created #549 to track this and to break up #515 into dedicated work on implementing Resources in the SDK.